contracts/factories/BridgeFactory.sol

```diff
@@ -142,24 +150,27 @@ contract BridgeFactory is IERC1155TokenReceiver, TieredOwnable {
 
     } else {
       revert("BridgeFactory#onERC1155Received: INVALID_TOKEN");
-    }
-
+    }  
+    
+    emit Deposit(_from, _data.readBytes32(0));
```

Comment:
Shouldn't we validate that `_salt` is not zero? It seems to me like an easy mistake that we can catch with little additional cost.

Reply:
It could be a mistake, but perhaps it would allow the bridge operator to have a different logic for tx hash? I was thinking it could be nice to let wallets/apps that aren't aware of the salt (they just send assets to the bridge) to still work.

Reply:
Doing a deposit without a salt increases a lot the risk of a user doing two equal deposits causing a collision, any wallet that deposits in that way would be putting those funds at risk (it could be manually recovered, but still).

Maybe one alternative is to use `redepositNonce` for those scenarios, it's not ideal and for sure it would be more expensive, but at least those transactions wouldn't fail and wouldn't be at risk of collision either.

Reply:
But couldn't the bridge operator use their own salt if the user didn't provide one? Bridge operator is the one using the salt to generate the nonce, they could use any nonce that hasn't been used in practice.

Reply:
All instances of bridged have to agree on the same salt in a deterministic manner, we could do something like a "deterministic fallback salt" like tx-hash + log-index, but I wanted to avoid complexity on the bridged side for determining the salt, I consider that critical because if different versions of bridged choose different salts for the same note it could lead to:

A) The note never reaches majority of signatures and thus is never valid.
B) Two different notes reach majority of signatures, causing inflation.

Reply:
That's a good points didn't think about indépendant bridged. Using the redeposit nonce seems like a good compromise (I'll rename it).